Automattic/woocommerce-payments
# Remove maybe_update_date_for_subscription function

Fixes #5425 

#### Changes proposed in this Pull Request

Removing `WC_Payments_Subscription_Service::maybe_update_date_for_subscription()`.
This function was added prior to releasing WCPay Subscriptions and was implementing features that are not supported. 

If a merchant has the WooCommerce Subscriptions extension installed, WCPay will use tokenized subscriptions giving more control over billing including trials and end-dates. Without this extension, WCPay Subscriptions does not support these features.

This code is limited to WCPay Subscriptions, the call to `self::get_wcpay_subscription_id( $subscription )` will only return a value if the subscription is not a tokenized subscription. Additionally, this code path will only run via wp-admin.

This is a dead code path and I've opted to remove rather than deprecate it as it was never used in the wild.


#### Testing instructions

1. Add some logging or breakpoints to the `WC_Payments_Subscription_Service::maybe_update_date_for_subscription` method.
2. Purchase a subscription with WCPay Subscriptions.
3. Attempt to update the Subscription schedule, See that the `Schedule` meta-box is not shown.
4. Try to update anyway to see if any of the to-be-removed code paths are touched

1. Install and active WC Subscriptions
2. Purchase a subscription with a trial period.
3. Update the subscription schedule.
4. Verify that the to-be-removed code paths are not touched.

![image](https://user-images.githubusercontent.com/57298/215654816-9cfb214b-0f96-441a-9425-fc2df3714587.png)





*

-------------------


**Post merge**
